doc/source/programs/gdal_vector_geom_op_set_type.rst

```diff
@@ -0,0 +1,146 @@
+.. _gdal_vector_geom_op_set_type_subsubcommand:
+
+================================================================================
+"gdal vector geom-op set-type" sub-sub-command
+================================================================================
+
+.. versionadded:: 3.11
+
+.. only:: html
+
+    Modify the geometry type of a vector dataset.
+
+.. Index:: gdal vector geom-op set-type
+
+Synopsis
+--------
+
+.. program-output:: gdal vector geom-op set-type --help-doc
+
+Description
+-----------
+
+:program:`gdal vector geom-op set-type` can be used to modify the geometry type
+of geometry fields at the layer and/or feature level.
+
+It can also be used as a step of :ref:`gdal_vector_pipeline_subcommand`.
+
+The following groups of options can be combined together:
+-  :option:`--multi` / :option:`--single`
+-  :option:`--linear` / :option:`--curve`
+-  :option:`--xy` / :option:`--xyz` / :option:`--xym` / :option:`--xyzm`
+
+Standard options
+++++++++++++++++
+
+.. include:: gdal_options/of_vector.rst
+
+.. include:: gdal_options/co_vector.rst
+
+.. include:: gdal_options/overwrite.rst
+
+.. include:: gdal_options/active_layer.rst
+
+.. option:: --active-geometry <ACTIVE-GEOMETRY>
+
+    Set the active geometry field from its name. When it is specified, only the
+    specified geometry field will be subject to the processing. Other geometry
+    fields will be not modified.
+    If this option is not specified, all geometry fields will be subject to the
+    processing.
+    This option can be combined together with :option:`--active-layer`.
+
+.. option:: --layer-only
+
+    Only modify the layer geometry type. A typical use case is if the layer
+    geometry type is set to unknown and the user knows that the layer only
+    contains geometry of a given type.
+
+.. option:: --feature-only
+
+    Only modify the geometry type of features.
+
+.. option:: --geometry-type <GEOMETRY-TYPE>
+
+   Change the geometry type to be one of
+   ``GEOMETRY``, ``POINT``, ``LINESTRING``, ``POLYGON``, ``MULTIPOINT``, ``MULTILINESTRING``,
+   ``MULTIPOLYGON``, ``GEOMETRYCOLLECTION``, ``CURVE``, ``CIRCULARSTRING``, ``COMPOUNDCURVE``,
+   ``SURFACE``, ``CURVEPOLYGON``, ``MULTICURVE``, ``MULTISURFACE``, ``POLYHEDRALSURFACE`` or ``TIN``.
+   ``Z``, ``M`` or ``ZM`` suffixes can be appended to the above values to
+   indicate the dimensionality.
+
+   This option is mutually exclusive with :option:`--multi`, :option:`--single`,
+   :option:`--linear`, :option:`--curve`, :option:`--xy`, :option:`--xyz`,
+   :option:`--xym`, :option:`--xyzm`.
+
+.. option:: --multi
+
+   Force geometries to MULTI geometry types. e.g. ``POINT`` ==> ``MULTIPOINT``.
+   This option is mutually exclusive with :option:`--single`.
+
+.. option:: --single
+
+   Force geometries to non-MULTI geometry types. e.g. ``MULTIPOINT`` ==> ``POINT``.
+   This option is mutually exclusive with :option:`--multi`.
```

Comment:
Indicate what happens if an input has > 1 component ? Do extra components get dropped or produce an error ?

Reply:
added: "MultiPoint or MultiLinestring with more than one component will be kept
   unmodified. MultiPolygon with more than one component will be merged into
   a (likely invalid according to OGC Simple Features rule) single polygon,
   merging all their rings."